javascripts/content_scripts/posting_button.js

```diff
@@ -1,131 +1,323 @@
+ * 
+ * @param  {Element}  target
+ * @return {Boolean}
+ */
+PrivlyButton.isTargetValid = function(target) {
+  return (target.nodeName === "TEXTAREA" || target.isContentEditable);
+};
+
+/**
+ * Get the most outside target element
+ * 
+ * @param  {Element} target
+ * @return {Element}
+ */
+PrivlyButton.getOuterTarget = function(target) {
+  while (target.parentNode && target.parentNode.isContentEditable) {
+    target = target.parentNode;
+  }
+  return target;
+};
+
+/**
+ * Get the privly button attached to the target element
+ * 
+ * @param  {Element} target
+ * @return {object} the privly button instance. return null if not attached.
+ */
+PrivlyButton.getAttachedButton = function(target) {
+  var container = target.parentNode;
+  var buttons = container.getElementsByClassName(privlyButtonClassName);
+  if (buttons.length === 0) {
+    return null;
+  } else {
+    return new PrivlyButton(target, buttons[0]);
+  }
+}
+
+function initEventListeners() {
+  function onTargetActivated(event) {
+    if (PrivlyButton.isTargetValid(event.target)) {
+      var target = PrivlyButton.getOuterTarget(event.target);
+      var button = PrivlyButton.getAttachedButton(target);
+      if (button === null) {
+        // this target does not has a privly button attached
+        button = new PrivlyButton(target);
       }
+      button.updatePosition();
+      button.attachAutoPosition();
+      button.show();
+      button.postponeHide();
     }
-  });
-
-  // The button will not disappear while it is hovered
-  span.addEventListener( "mouseover", function(){
-      hovered = true;
-      div.style.opacity = "1";
-      clearTimeout(timeout);
-    });
-
-  span.addEventListener( "mouseout", function(){
-      hovered = false;
-      div.style.opacity = "0.7";
-      timeout = setTimeout(fadeOut, timeoutLength);
-    });
-
-  // Clicking the button will send a message to posting_process.js to create new
-  // Privly message using Message application
-  span.addEventListener( "click", function() {
-
-    // Check if there is no pending post and if the button has been triggered
-    // i.e. the opacity is 0.7
-    if (active && !pendingPost &&  (getComputedStyle(div).getPropertyValue("opacity") > 0)) {
-      chrome.runtime.sendMessage({ask: "newPost"}, function(response) {});
-      privlyUrlReceiptNode = context;
-    } else {
-      chrome.runtime.sendMessage({ask: "showNotification"}, function(response) {});
+  }
+  function onTargetDeactivated(event) {
+    if (PrivlyButton.isTargetValid(event.target)) {
+      var target = PrivlyButton.getOuterTarget(event.target);
+      var button = PrivlyButton.getAttachedButton(target);
+      if (button !== null) {
+        button.detachAutoPosition();
+        // If we click the button, onBlur will be triggered before onClick,
+        // so we have to shortly postpone the hiding process.
+        button.postponeHide(HIDE_DELAY_SHORT);
+      }
     }
-  });
+  }
+  // would onMouseDown be a better choice?
```

Comment:
Wow, funny edge case. You could add a check to `onTargetDeactivated` for whether the current focus is the button. When the button blurs back the button might be added back if the form element was subsequently focused which would cause a shimmer? Regardless, I am hesitant to add too much complexity for this edge case.